File: path/to/project/database/insights.py
from pymongo import MongoClient

# Connecting to MongoDB
def connect_db():
    client = MongoClient('mongodb://localhost:27017/')
    return client['gameplay_analyzer']

# Inserting insights and statistics
def insert_insights(insight_data):
    db = connect_db()
    collection = db['insights']
    result = collection.insert_one(insight_data)
    return result.inserted_id

# Retrieving insights and statistics
def retrieve_insights():
    db = connect_db()
    collection = db['insights']
    return list(collection.find({}))

File: path/to/project/database/insights.js
const { MongoClient } = require('mongodb');

// Connecting to MongoDB
const url = 'mongodb://localhost:27017';
const dbName = 'gameplay_analyzer';

async function connectDb() {
    const client = new MongoClient(url);
    await client.connect();
    return client.db(dbName);
}

// Inserting insights and statistics
async function insertInsights(insightData) {
    const db = await connectDb();
    const collection = db.collection('insights');
    const result = await collection.insertOne(insightData);
    return result.insertedId;
}

// Retrieving insights and statistics
async function retrieveInsights() {
    const db = await connectDb();
    const collection = db.collection('insights');
    return await collection.find({}).toArray();
}